### PR TITLE
gh-104146: Purge dead code from Argument Clinic

### LINKED ISSUE #104146: Remove dead code from clinic.py
There's a bit of unused code in clinic.py. For example, the code that manipulated `second_pass_replacements` was removed in 2015 (commit 0759f84d6260bad1234b802212e73fdc5873d261, gh-67688 (bpo-23500)), but the variable and a branch depending on it still persists.

Suggesting to remove all unused code in order to make clinic more readable and maintainable.

### Linked PRs
* gh-104147
* gh-104214
* gh-104627
* gh-104680
* gh-106358
* gh-107555
* gh-107608
* gh-107627
* gh-107632
* gh-119162

### PR DESCRIPTION
**Pull request by erlend-aasland**
The following local variables were assigned but never used:

- line 551:  result
- line 1341: groups
- line 1431: default_return_converter
- line 1529: ignore_self
- line 1809: input_checksum
- line 4224: new'

* Issue: gh-104146

### CODE REVIEW

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -4245,8 +4237,7 @@ def directive_version(self, required):
             fail("Insufficient Clinic version!\n  Version: " + version + "\n  Required: " + required)
 
     def directive_module(self, name):
-        fields = name.split('.')
-        new = fields.pop()
+        fields = name.split('.')[1:]
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py]: pop() removes the last item in the list, no?

```suggestion
        fields = name.split('.')[:-1]
```

**Comment by AlexWaygood** [comment]
Would you be interested in possibly running pyflakes on Tools/clinic/ in CI, to catch this kind of thing in the future? It's a linter that tries extremely hard to have 0 false positives, and didn't have any hits on Tools/clinic/ other than the ones that are being fixed here

File: Tools/clinic/clinic.py (hunk 1 above)
Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: Good catch. Yes, `[1:]` is definitely wrong. Strangely, `make clinic` does not complain or generate incorrect.

**Comment by erlend-aasland** [comment]
> Would you be interested in possibly running pyflakes on Tools/clinic/ in CI, to catch this kind of thing in the future? It's a linter that tries extremely hard to have 0 false positives, and didn't have any hits on Tools/clinic/ other than the ones that are being fixed here

_I_ think that's a good idea. Others might disagree :)

**Comment by AlexWaygood** [comment]
I can open an issue and we can see if anybody objects!

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -4246,7 +4238,7 @@ def directive_version(self, required):
 
     def directive_module(self, name):
         fields = name.split('.')
-        new = fields.pop()
+        _ = fields.pop()
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py]: I'd personally find this more idiomatic:

```py
fields = name.split[:-1]
```

But this is also fine :)

**Review by AlexWaygood — APPROVED** [review]

File: Tools/clinic/clinic.py (hunk 2 above)
Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: I agree! I was just trying to keep the diff down, but let's exploit the moment and make this nicer :) Feel free to update the PR and land it; I'm on mobile for the rest of the day.

**Pull request merged**